src/main/java/mate/jdbc/service/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,125 @@
+package mate.jdbc.service;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    private static final int PARAMETER_INDEX_OF_NAME_ID = 1;
+    private static final int PARAMETER_INDEX_OF_COUNTRY = 2;
+    private static final int PARAMETER_INDEX_OF_ID = 3;
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertManufacturerQuery =
+                "INSERT INTO manufacturers(name, country) VALUES(?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufacturerStatement =
+                        connection.prepareStatement(insertManufacturerQuery,
+                             Statement.RETURN_GENERATED_KEYS)) {
+            createManufacturerStatement.setString(PARAMETER_INDEX_OF_NAME_ID,
+                    manufacturer.getName());
+            createManufacturerStatement.setString(PARAMETER_INDEX_OF_COUNTRY,
+                    manufacturer.getCountry());
+            createManufacturerStatement.executeUpdate();
+            ResultSet generatedKey = createManufacturerStatement.getGeneratedKeys();
+            if (generatedKey.next()) {
+                Long id = generatedKey.getObject(PARAMETER_INDEX_OF_NAME_ID, Long.class);
+                manufacturer.setId(id);
+            }
+            return manufacturer;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't insert manufacturer "
+                    + "to DataBase " + manufacturer, ex);
+        }
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String getManufacturerQuery = "SELECT * FROM manufacturers WHERE id = ? "
+                + "AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturerStatement =
+                        connection.prepareStatement(getManufacturerQuery)) {
+            getManufacturerStatement.setLong(PARAMETER_INDEX_OF_NAME_ID, id);
+            ResultSet resultSet = getManufacturerStatement.executeQuery();
+            Manufacturer manufacturer = null;
+            while (resultSet.next()) {
```

Comment:
```suggestion
            if (resultSet.next()) {
```
you can replace with if because you should only one row of results